lib/mxlib/matrix.js

```diff
@@ -322,7 +319,19 @@ var setJoinRule = function(client, roomId, joinRule) {
         roomId, "m.room.join_rules", {
             join_rule: joinRule
         }, ""
-    );
+    ).then(function() {
+        return roomId;
+    });
+};
+
+var setRoomHistoryVisibility = function(client, roomId, history_visibility) {
+    return client.sendStateEvent(
+        roomId, "m.room.history_visibility", {
+            "history_visibility": history_visibility
+        }, ""
+    ).then(function() {
+        return roomId;
```

Comment:
Its purely to make the chaining of the deferreds above slightly nicer. In the end it probably doesn't make that much difference, since we would just have to move the `.then(function() {return roomId;})` to the call sites above.